api/app/controllers/application_controller.rb

```diff
@@ -1,4 +1,26 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
+  include JsonWebToken
+
+  def auth_header
+    request.headers['Authorization']
+  end
+
+  def decoded_token
+    return unless auth_header
+
+    token = auth_header.split(' ')[1]
+    decode_token(token)
+  end
+
+  def current_user
+    return unless decoded_token
+
+    auth_token = AuthToken.find_by_id(decoded_token[0]['id'])
+
+    return unless auth_token
+
+    auth_token.user
+  end
```

Comment:
Should this logic exist in the `ApplicationController`, or should that be refactored elsewhere?

Reply:
current_user is good at application level because you want to handle this logic everywhere.

Make sure here that you handle the use cases of tokens not existing properly (returning nil or an object that is like a NotLoggedInUser if you want more of an OO style)

Reply:
I decided to move this into the `GraphqlController` because I don't really want any auth to happen outside of my graphql route. I may refactor this later if the api grows beyond a simple graphql endpoint.